run_unittests.py

```diff
@@ -2838,6 +2839,43 @@ def test_cflags_cross_environment_pollution(self):
         self.assertNotIn('-DBUILD_ENVIRONMENT_ONLY', compdb[0]['command'])
 
 
+class PythonTests(BasePlatformTests):
+    '''
+    Tests that verify compilation of python extension modules
+    '''
+    def test_versions(self):
+        if self.backend is not Backend.ninja:
+            raise unittest.SkipTest('Skipping python tests with {} backend'.format(self.backend.name))
+
+        testdir = os.path.join(self.src_root, 'test cases', 'python', '1 extmodule')
+
+        # No python version specified, this will use meson's python
+        self.init(testdir)
+        self.build()
+        self.run_tests()
+        self.wipe()
+
+        # When specifying a known name, (python2 / python3) the module
+        # will also try 'python' as a fallback and use it if the major
+        # version matches
+        self.init(testdir, ['-Dpython=python2'])
+        self.build()
+        self.run_tests()
+        self.wipe()
```

Comment:
I ran this test on Windows with only Python 3 installed. It should fail but instead it gets skipped. It should check the version of `python.exe` and check that it either fails with an error (if it is Python 3) or succeeds (if Python 2).

Reply:
No, `self.init(testdir)` throws a skipexception both on lines 2852 and 2860.

Reply:
OK, well I'll take a look, can you please provide a review of the API anyway?

Reply:
Also, if you don't mind giving some more details about your setup on Windows to make it easier for me to reproduce that would be appreciated :)

Reply:
It's literally only Windows 10 VM from modern.ie + Python 3.newest 64 bit.